wordpress-mobile/WordPress-iOS
# Add warning screen when deleting site

This PR fixes #6629 by adding an extra screen before deleting a site.

![result](https://cloud.githubusercontent.com/assets/154014/23315902/d3e492f8-fa8d-11e6-9b77-2d7b5e16ac4b.png)

The text on this new screen was copied directly from the Calypso screens provided by @rachelmcr in #6629 

Most of the same delete site functionality that existed on the `SiteSettingsViewController` has been moved into the new `DeleteSiteViewController`. In addition, the `DeleteSiteViewController` was created as a grouped UITableVC to match the style of `StartOverViewController` in order to maintain consistency. The new workflow looks like this:

<img width="1232" alt="napkin 39 02-24-17 12 30 57 pm" src="https://cloud.githubusercontent.com/assets/154014/23316042/7beb1b48-fa8e-11e6-9afe-695821338a4f.png">

To test:
1. Press the "Delete Site" button in Site Settings. 
2. On a site without upgrades, you should be taken to the new Delete Site screen. 
2a. Attempt to delete the site.
2b. Attempt to contact support
3. On a site with upgrades, the upgrades check should present a Premium Upgrades alert view. Attempt to navigate to the purchases web view.

Also, be sure to text this on the iPads as well as iPhone.

Needs review... @frosty would you mind? :-)
